src/shared/services/auth/auth.js

```diff
@@ -80,41 +83,37 @@ function saveRefreshToken(refreshToken) {
  */
 async function refreshTokens() {
   const { authEndpoint, realm, clientId } = configuration.keycloak;
-  const token = storage.getItem(REFRESH_TOKEN_KEY);
+  const refreshToken = storage.getItem(REFRESH_TOKEN_KEY);
+  const decodedRefreshToken = accessTokenParser(refreshToken);
+
+  if (decodedRefreshToken.expiresAt * 1000 < Date.now()) return;
```

Comment:
You are depending on local system time here. When the local system time is wrong, the application will not work. Why not just try the access token and if the token is not working, try to refresh the token?

Reply:
Your suggestion makes sense, but the current behavior is that users are authenticated as long as they have an access token that is not expired. 

The Keycloak adapter also uses Date.now() under the hood to check for token expiry. There's an option to provide a time skew (effectively an estimate of the time difference between client and server), but I can't think of a way to effectively estimate it. 

I can set a default time skew of e.g. 2 seconds